rllib/env/wrappers/atari_wrappers.py

```diff
@@ -208,28 +208,60 @@ def reset(self, **kwargs):
 
 @PublicAPI
 class MaxAndSkipEnv(gym.Wrapper):
-    def __init__(self, env, skip=4):
-        """Return only every `skip`-th frame"""
-        gym.Wrapper.__init__(self, env)
-        # most recent raw observations (for max pooling across time steps)
-        self._obs_buffer = np.zeros((2,) + env.observation_space.shape, dtype=np.uint8)
+    """Performs frame skipping on env with max'ing observations over last 2 frames.
+
+    Note that when using this wrapper with Atari envs, your `frameskip` gym.make()
+    setting should probably be set to 1, as you probably don't want to have two
+    frame skipping wrappers active.
+
+    Within the skipped frames, this wrapper will return the maximum individual-dim
+    observation values over the last two frames.
+    For example, if observation_space = Box(0, 255, (1, ), uint8) and we skip over the
+    4 observations: [9] [4] [100] and [59], we would return [100] (max over the last
+    two skipped frames).
+    """
+    def __init__(self, env, skip: int = 4, max_over: int = 2):
+        """Initializes a MaxAndSkipEnv instance.
+
+        Args:
+            skip: The number of frames to skip with each `step()` call. Note that
+                after `reset()` no frames are skipped.
+            max_over: The number of the last n frames (within the skipped ones) that
+                we max the returned observations over. Each dim in the observation
+                space is maxed over individually, not changing the shape (or dtype)
+                of the observation space.
```

Comment:
can we add documentation that skip must be greater than or equal to max_over, then provide an example of what happens when skip is greater than max over.